src/Platform/Microsoft.Testing.Platform/Hosts/TestHostControllersTestHost.cs

```diff
@@ -129,6 +131,12 @@ protected override async Task<int> InternalRunAsync()
                 dataConsumersBuilder.Add(dataConsumerDisplay);
             }
 
+            // We register the DotnetTestDataConsumer as last to ensure that it will be the last one to consume the data.
```

Comment:
Do we ensure that registration order is matching reception order?

Reply:
yep

Reply:
here anyway is less than important anyway we wait that all queue are empty before to move on drain.